multi-agent/lib/mas/core/field_hints.py

```
@@ -356,7 +356,35 @@ def to_hints(self) -> Dict[str, Any]:
+    - ``read_only=False`` → field is user-configurable (per-user overlay).
+
+    Fields without this hint default to read-only when served via
+    ``get_builtin_schema()``.  For non-built-in resources the hint is
+    ignored and all fields remain editable.
+    """
+    read_only: bool = Field(
+        default=True,
+        description="When True, the field cannot be edited by users on built-in resources"
+    )
+
+    def model_dump(self, **kwargs) -> Dict[str, Any]:
```

Ruff (0.15.21)

[warning] Missing type annotation for `**kwargs`

(ANN003)